test/module/shared_model/validators/commands_validator_test.cpp

```diff
@@ -0,0 +1,316 @@
+  iroha::protocol::SetAccountQuorum command;
+
+  command.set_account_id(account_id);
+  command.set_quorum(quorum);
+
+  return command;
+}
+
+iroha::protocol::TransferAsset generateTransferAsset(
+    std::string src_account_id,
+    std::string dest_account_id,
+    std::string asset_id,
+    std::string description,
+    std::string amount) {
+  iroha::protocol::TransferAsset command;
+
+  command.set_src_account_id(src_account_id);
+  command.set_dest_account_id(dest_account_id);
+  command.set_asset_id(asset_id);
+  command.set_description(description);
+
+  auto iroha_amount = iroha::Amount::createFromString(amount).value();
+  auto proto_amount = command.mutable_amount();
+  proto_amount->set_precision(iroha_amount.getPrecision());
+  auto proto_amount_value = proto_amount->mutable_value();
+  auto uint64s = iroha_amount.to_uint64s();
+  proto_amount_value->set_first(uint64s.at(0));
+  proto_amount_value->set_second(uint64s.at(1));
+  proto_amount_value->set_third(uint64s.at(2));
+  proto_amount_value->set_fourth(uint64s.at(3));
+
+  return command;
+}
+
+iroha::protocol::AppendRole generateAppendRole(std::string account_id,
+                                               std::string role_name) {
+  iroha::protocol::AppendRole command;
+
+  command.set_account_id(account_id);
+  command.set_role_name(role_name);
+
+  return command;
+}
+
+iroha::protocol::CreateAccount generateCreateAccount(std::string account_name,
+                                                     std::string domain_id,
+                                                     std::string main_pubkey) {
+  iroha::protocol::CreateAccount command;
+
+  command.set_account_name(account_name);
+  command.set_domain_id(domain_id);
+  command.set_main_pubkey(main_pubkey);
+
+  return command;
+}
+
+iroha::protocol::CreateRole generateCreateRole(
+    std::string role_name,
+    std::vector<iroha::protocol::RolePermission> permissions) {
+  iroha::protocol::CreateRole command;
+
+  command.set_role_name(role_name);
+  for (auto permission : permissions) {
+    command.add_permissions(permission);
+  }
+
+  return command;
+}
+
+iroha::protocol::GrantPermission generateGrantPermission(
+    std::string account_id, iroha::protocol::GrantablePermission permission) {
+  iroha::protocol::GrantPermission command;
+
+  command.set_account_id(account_id);
+  command.set_permission(permission);
+
+  return command;
+}
+
+iroha::protocol::RevokePermission generateRevokePermission(
+    std::string account_id, iroha::protocol::GrantablePermission permission) {
+  iroha::protocol::RevokePermission command;
+
+  command.set_account_id(account_id);
+  command.set_permission(permission);
+
+  return command;
+}
+
+using namespace iroha::protocol;
+using namespace shared_model;
+
+/**
+ * @given transaction made of commands with valid fields
+ * @when commands validation is invoked
+ * @then answer has no errors
+ */
+TEST(CommandsValidatorTest, StatelessValidTest) {
```

Comment:
This test looks complicated. Maybe better to separate it?

Reply:
Into 12 tests for each command??

Reply:
It can be improved in the following way:
1. Extract valid consts to the global scope
2. Set the default params of generate* with those vars
3. Clean up the code

And writing separate test for each method I think not so good idea